### DISCUSSION TITLE
Top Sources missing direct traffic on default view

### DISCUSSION THREAD

**Discussion by andrewholgate** [plausible/analytics#1539] [Anything Else]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Describe the bug

On the default view of the Top Sources, the "Direct / None" traffic is not displayed. It is only shown when the view is expanded.

### Expected behavior

The default view of the Top Sources would display all of the sources, including "Direct / None".

### Screenshots

![Screenshot_20211215-064404](https://user-images.githubusercontent.com/3862180/146130427-4a8a1c16-5e5f-46b4-8f6a-f9d6c7fae2d1.png)

**Update:** "Direct / None" traffic is now shown in the main view of Top Sources as well!

**Comment by metmarkosaric** [comment]
thanks for reporting that @andrewholgate! this is intentional and there's no plan to change it. "direct/none" traffic can be found under "details" tab in Top Sources report. when we were building this we were testing both versions (direct in default list and the current way). turns out most sites have a lot of direct traffic and the community didn't find the "direct/none" number actionable plus it skewed the comparison we do in the default view so we left it out in the "details" only. hope that makes sense, thanks!

**Reply by Vrq** [reply]
In my opinion, it would be better to have the preference setting to either show or hide direct by default.

**Reply by metmarkosaric** [reply]
Thanks for the feedback! We focus on being a simple to use tool which means that we prefer to introduce features that work for the majority. This also means that if something is not good enough for the majority we prefer not to have it in settings as an option either as that would then lead to having hundreds of different options in settings which means we will end up like some kind of monster that GA is today.

**Comment by andrewholgate** [comment]
Ah yes, I got the impression it was an intentional decision to remove it. While I would find it useful to have it visible on the main view, it seems like I might be in the minority.

**Reply by metmarkosaric** [reply]
yeah we had a poll in social media about a year and a half ago asking about this and this was the decision made by the majority. direct is still there a click away for those who need it

**Reply by andrewholgate** [reply]
I am more than happy to live with this decision. It's not really an issue, just more of a minor inconvenience (for us minority).

**Comment by ingejanse** [comment]
I'm a new user of Plausible (hi!), and I also find it weird that 'direct' is missing. It's not really a source, but then again, url type in / bookmark in a way is. Furthermore, I kept being confused by the fact that 'direct' is visible in details (these are the details of sources, so if you are consistent, then you should also exclude it there). 

My context is this: I use 'sources' to give me an idea where my audience comes from. 'Direct' is a perfectly plausible (pun!) parameter here for me. Any chance on this?

**Reply by Vrq** [reply]
I also agree on that - because otherwise the app is very inconsistent. And to get the full picture, we also need to see the 'dark' traffic.

**Reply by metmarkosaric** [reply]
hi! and thanks for the feedback! when we were building this, we had both versions and asked the community about it and the majority didn't want direct to be displayed there as it's not actionable and skews the comparison. that's why we put it under details so it's only a click away for those that want to see direct. there are no current plans to change this.

**Reply by Vrq** [reply]
Hi Marko, 

We understand that decision, but maybe you could add a flag in the settings for that?

I know that you don't want to have the settings too bloated, however, for some power-users it would be nice to have an "Advanced settings" section.

**Reply by metmarkosaric** [reply]
perhaps we can do that in the future if we change the approach and start having advanced settings for niche purposes. until now we're focused on being simple to use and minimal, so we'll need to consider how having settings for things only a few people use will affect the rest of Plausible and what people love about Plausible currently

**Reply by Vrq** [reply]
To be honest, I don't see any impact for existing users if you would add a tab hidden in the settings to open advanced settings and a disclaimer that they are only for power users

**Reply by metmarkosaric** [reply]
Just having a page with a disclaimer "for power users only" is very different from the whole Plausible approach until now but again perhaps something we can consider in the future if we want to change things

**Reply by ingejanse** [reply]
> hi! and thanks for the feedback! when we were building this, we had both versions and asked the community about it and the majority didn't want direct to be displayed there as it's not actionable and skews the comparison. that's why we put it under details so it's only a click away for those that want to see direct. there are no current plans to change this.

Okidoki. One last attempt, then I'll be silent about it forever and ever! 

1: clicking on 'direct' is perfectly actionable, right? I'm much interested to learn where they have entered the site, where people come from, et cetera. 
2: this skewing of comparisons also works the other way; now it looks for my team members that Google is the first and LinkedIn the second source of traffic (they're not the people how like to see details, they just want a general overview). I would really like them to realize that direct is second, and how much it differs from the first and third source. IE I (and they) would be much helped if direct is in this overview.
3: the inconsistency of having direct not in the general overview but nevertheless in the details is very hard to swallow for people like me how like to order their books by colour and live in venn diagrams :)

**Reply by andrewholgate** [reply]
@metmarkosaric  I agree with the others on this topic, the decision to hide "Direct / none" creates an inconsistency and user experience issue to many user I would say. I fear that if you have your users vote on little details like this, you might be missing out on the bigger picture of ensuring a consistent product.

**Reply by metmarkosaric** [reply]
thanks for all the feedback, you've been heard. we'll consider if there's a need to change the approach and will update this thread if there are any news on the topic.

**Reply by ingejanse** [reply]
Thanks! I'll refrain to re-ordering my books for now. I didn't categorize them on height yet :)

**Comment by mmasouras** [comment]
Between this ticket and #814 there are now at least 5 people asking for this. 

I am not sure where that survey is done, and what is "the community". I run my business using plausible and I was never asked about this. I don't think having to monitor all social media platforms for when plausible will run a survey is a good way to understand the importance of a feature, it sounds incomplete and ad-hoc. It also is a bad signal as it severely biases the sample your are using to make a decision. 

Finally, I haven't heard a concrete explanation of how having Direct shown harms the user experience, and why showing 5 visitors from Yandex or whatever is preferable.

**Reply by metmarkosaric** [reply]
Thanks for the feedback, you've been heard. There are no short term plans to change this as we're focused on more popular requests such as the GA data import but if or when things change we will update this post.

**Reply by andrewholgate** [reply]
@metmarkosaric can you please provide a link to this particular survey? I'd also like to understand how I am able to participate in all future survey's as well.

**Reply by metmarkosaric** [reply]
when we do surveys, they would be on our social media ([twitter](https://twitter.com/plausiblehq) and [mastodon](https://fosstodon.org/@plausible)) or here on github

**Reply by andrewholgate** [reply]
I could not track it down, however this was the closest I could find (which if extrapolated, suggests that users would prefer to know about this "dark traffic" - https://twitter.com/PlausibleHQ/status/1263141632209235969

Can I suggest that a consistent and inclusive method for survey is used for the future? You have the email addresses of your paying users, who I am sure would want to be informed when there are decisions being voted on that impact the product.

**Reply by metmarkosaric** [reply]
we do surveys occasionally. we've done 3-4 until now. the purpose of them is to guide and inform possible decisions so there is no need to email everyone that uses Plausible about it (we don't send any broadcast emails to all customers even for other purposes). we may create a survey mailing list that people that are interested in this can subscribe to in the future but for the moment the best place is our social media. and again, if our plans for the inclusion of direct/none on the main screen change, we will update this thread. thanks again for the feedback!

**Comment by metmarkosaric** [comment]
**Update:** "Direct / None" traffic is now shown in the main view of Top Sources as well!
